multicast/src/main/kotlin/com/dropbox/flow/multicast/ChannelManager.kt

```diff
@@ -43,170 +48,253 @@ internal class ChannelManager<T>(
      * Instead, it is kept open and if a new downstream shows up that causes us to restart the flow,
      * it will receive values as well.
      */
-    private val piggybackingDownstream: Boolean = false,
+    piggybackingDownstream: Boolean = false,
     /**
      * Called when a value is dispatched
      */
-    private val onEach: suspend (T) -> Unit,
-    /**
-     * Called when the channel manager is active (e.g. it has downstream collectors and needs a
-     * producer)
-     */
-    private val onActive: (ChannelManager<T>) -> SharedFlowProducer<T>
-) : StoreRealActor<ChannelManager.Message<T>>(scope) {
-    private val buffer = Buffer<T>(bufferSize)
-    /**
-     * The current producer
-     */
-    private var producer: SharedFlowProducer<T>? = null
+    onEach: suspend (T) -> Unit,
 
-    /**
-     * Tracks whether we've ever dispatched value or error from the current producer.
-     * Reset when producer finishes.
-     */
-    private var dispatchedValue: Boolean = false
+    upstream: Flow<T>
+) {
 
-    /**
-     * List of downstream collectors.
-     */
-    private val channels = mutableListOf<ChannelEntry<T>>()
-
-    override suspend fun handle(msg: Message<T>) {
-        when (msg) {
-            is Message.AddChannel -> doAdd(msg)
-            is Message.RemoveChannel -> doRemove(msg.channel)
-            is Message.DispatchValue -> doDispatchValue(msg)
-            is Message.DispatchError -> doDispatchError(msg)
-            is Message.UpstreamFinished -> doHandleUpstreamClose(msg.producer)
-        }
+    private val actor = Actor<T>(scope, bufferSize, piggybackingDownstream, onEach, upstream)
+
+    suspend operator fun plusAssign(channel: SendChannel<Message.Dispatch.Value<T>>) =
```

Comment:
i find this confusing vs sending an explicit message.
my vote is to keep the previous way of explict message passing instead of syntactic sugar

Reply:
I'd like to hide the fact the a ChannelManager is an actor, that's just a vector for mistakes later. For example someone can misuse a channel manager to inject incoming `DispatchValue`s that are not coming from the upstream (as `simple` test is doing)

Reply:
isn't it weird for someone to send a messge to channel manager w/o knowing what it does?
Also the message is a sealed class so it is not much different from being a sealed class vs being accessible methods on it ?